### ISSUE TITLE
@Suppress("Detekt.ruleset") feature

### ISSUE THREAD

**Issue by vanniktech** [arturbosch/detekt#766]
I have this function that I use for testing quite often:

```kotlin
inline fun <reified T> assertThrows(block: () -> Unit): AbstractThrowableAssert<*, out Throwable> {
  try {
    block()
  } catch (e: Throwable) {
    if (e is T) {
      return assertThat(e)
    } else {
      throw e
    }
  }

  throw AssertionError("Expected ${T::class.simpleName}")
}
```

Detekt is going wild on it and flagging all of these: `TooGenericExceptionCaught`, `InstanceOfCheckForException`, `RethrowCaughtException`.

Instead of `@Suppress("Detekt.TooGenericExceptionCaught", "Detekt.InstanceOfCheckForException", "Detekt.RethrowCaughtException")` - I know `Detekt` is optional but I like to see where things are coming from - it would be cool if I could just do `@Suppress("Detekt.exceptions")`.

**Comment by schalkms** [comment]
Looks good to me

**Comment by arturbosch** [comment]
Makes sense to suppress whole rulesets. Unfortunately detekt needs heavy chances to support this (suppression is checked for `Finding.id`, we do not have information about the rule set id anymore).
How about you just add them to your baseline xml file?

**Comment by vanniktech** [comment]
I don't have a baseline file since I'm trying to stay at 0 warnings / errors. Can't we rework that part in detekt?

**Comment by Mauin** [comment]
Something that should be easier to start with is excluding the entire function for all detekt rules. Just something like `Suppress("Detekt")`. Adding support for entire rulesets in the future might make sense as well.

**Comment by vanniktech** [comment]
I'd still like to get other things flagged, just not the exception part.

**Comment by arturbosch** [comment]
Thats a good feature after 1.0. That won't change the user interfaces but is a good candidate for a major version :D

**Comment by arturbosch** [comment]
Ok, new idea for this.

currently the suppression logic is within `detekt-api` which already feels kinda wrong.
Also we are checking for suppresion inside the report function:
```kotlin
	override fun report(finding: Finding) {
		val ktElement = finding.entity.ktElement
		if (ktElement == null || !ktElement.isSuppressedBy(finding.id, finding.issue.aliases)) {
			_findings.add(finding)
		}
	}
```
If we move the logic inside `detekt-core`, more specific to `DetektFacade` we can make use of the 
ruleSetId to findings sorting for this:
```kotlin
val findings = HashMap<String, List<Finding>>()
```

**Comment by vanniktech** [comment]
> If we move the logic inside detekt-core, more specific to DetektFacade we can make use of the
ruleSetId to findings sorting for this:

Sounds breaking to me. And if so we should do it before 1.0.0. I've looked at it quickly but could not quite figure the bits out so if anyone else wants to take a stab at it - please do!
